File: baekjoon_OnlineJudge/LearnCpp/10828.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main() {
	int N;
	cin >> N;
	stack<int> stack;

	for (int i = 0; i < N; i++) {
		string A;
		cin >> A;
		if (A == "push") {
			int num;
			cin >> num;
			stack.push(num);
		}
		else if (A == "pop") {
			if (!stack.empty()) {
				cout << stack.top() << '\n';
				stack.pop();
			}
			else {
				cout << "-1" << '\n';
			}
		}
		else if (A == "size") {
			cout << stack.size() << endl;
		}
		else if (A == "empty") {
			if (stack.empty()) {
				cout << "1" << '\n';
			}
			else {
				cout << "0" << '\n';
			}
		}
		else if (A == "top") {
			if (!stack.empty()) {
				cout << stack.top() << '\n';
			}
			else {
				cout << "-1" << '\n';
			}
		}
	}
}

File: BOJ/Algorithms/DataStructure/10828.py
import sys ; input = lambda : sys.stdin.readline().rstrip()

stack = []

N = int(input())

for _ in range(N):
    data = input().split()

    if data[0] == 'push':
        stack.append(data[1])
    elif data[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif data[0] == 'size':
        print(len(stack))
    elif data[0] == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif data[0] == 'top':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])
